packages/shared/src/components/feeds/FeedContainer.tsx

```diff
@@ -186,6 +203,25 @@ export const FeedContainer = ({
     return feedNameToHeading[feedName] ?? '';
   }, [feeds, feedName, router.query.slugOrId]);
 
+  const { getMarketingCta, clearMarketingCta } = useBoot();
+  const marketingCta = getMarketingCta(MarketingCtaVariant.FeedBanner);
+  const { shouldShow } = useShowUpload();
+  const { onUpload, status } = useUploadCv({
+    onUploadSuccess: () => clearMarketingCta(marketingCta.campaignId),
+  });
+  const justUploaded = status === 'success';
+  const shouldShowBanner = (!!marketingCta && shouldShow) || justUploaded;
+
+  const clearMarketingCtaRef = useRef(clearMarketingCta);
+  clearMarketingCtaRef.current = clearMarketingCta;
+
+  useEffect(() => {
+    // when the user has uploaded their cv already, but marketing tool does not know it
+    if (!!marketingCta && !shouldShow && !justUploaded) {
+      clearMarketingCtaRef.current(marketingCta.campaignId);
+    }
+  }, [marketingCta, shouldShow, justUploaded]);
```

Comment:
Didn't we have some hook to manage marketingCTA internal and via app already?

Reply:
I wanted to do that too initially, but they are not inherently present in all cases. Let me verify that end again.